pservice/pdo/pservice/pdo_enclave.py

```diff
@@ -0,0 +1,273 @@
+    if missing_keys:
+        raise \
+            ValueError(
+                'PDO enclave config file missing the following keys: '
+                '{}'.format(
+                    ', '.join(sorted(list(missing_keys)))))
+
+    if not _ias:
+        _ias = \
+            ias_client.IasClient(
+                IasServer = config['ias_url'],
+                SpidCert = config['spid_cert_file'],
+                Spid = config['spid'],
+                HttpsProxy = config.get('https_proxy',""))
+
+    if not _pdo:
+        signed_enclave = __find_enclave_library(config)
+        logger.debug("Attempting to load enclave at: %s", signed_enclave)
+        _pdo = enclave.pdo_enclave_info(signed_enclave, config['spid'])
+        logger.info("Basename: %s", get_enclave_basename())
+        logger.info("MRENCLAVE: %s", get_enclave_measurement())
+
+    sig_rl_updated = False
+    while not sig_rl_updated:
+        try:
+            update_sig_rl()
+            sig_rl_updated = True
+        except (SSLError, Timeout, HTTPError) as e:
+            logger.warning("Failed to retrieve initial sig rl from IAS: %s", str(e))
+            logger.warning("Retrying in 60 sec")
+            time.sleep(60)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def shutdown():
+    global _pdo
+    global _ias
+    global _sig_rl_update_time
+    global _epid_group
+
+    _pdo = None
+    _ias = None
+    _sig_rl_update_time = None
+    _epid_group = None
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def get_enclave_measurement():
+    global _pdo
+    return _pdo.mr_enclave if _pdo is not None else None
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def get_enclave_basename():
+    global _pdo
+    return _pdo.basename if _pdo is not None else None
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def create_secret(len): return enclave.create_sealed_secret(len)
+
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def unseal_secret(secret): return enclave.unseal_secret(secret)
+
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def generate_enclave_secret(enclave_sealed_data, sealed_secret, enclave_id, contract_id, opk, enclave_key):
+ return enclave.generate_enclave_secret(enclave_sealed_data, sealed_secret, enclave_id, contract_id, opk, enclave_key)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def create_signup_info(nonce):
```

Comment:
Do we actually need all of this signup data stuff? The pservice enclave isn't actually registered with the ledger, unlike the eservice...

Reply:
It is not being used to register with enclave, just to create the enclave data. I can change the method names.